### DISCUSSION TITLE
Custom view plugin

### DISCUSSION THREAD

**Discussion by Thomas-debug-creator** [nasa/openmct#7482] [Community Q&A] [answered]
I want to develop and install my own plugins for my OpenMCT App. For instance, similar to the ones in [this project](https://www.fhnw.ch/plattformen/erc-rover/mission-control-center/), I want to add a custom way of representing the wheels orientation. I tried following [this issue](https://github.com/nasa/openmct/issues/3305) with Vue.js and webpack but didn't manage to make it work.

How would you recommend going about that? More generally, what do you recommend to implement your own visualization panels/widgets?

Thanks a lot!

**Comment by ozyx** [comment]
Hi @Thomas-debug-creator ,

Sorry for the delayed response. Have you gotten any further with this?

I have been working on an example plugin repo that just needs a good final pass before I can share. In the meantime, is it possible for you to share your code?

Cheers,
Jesse

**Comment by AdamElliott-AB** [comment]
@ozyx an example component using Vue would be very helpful. I have a custom vue component that I have used with an earlier version of OpenMCT (< 3.0.0) but am having trouble getting it upgraded to work with the latest version. I was hoping for some example like https://github.com/nasa/openmct-hello but using vue and working with the latest version of OpenMCT

**Reply by ozyx** [reply]
Hi @AdamElliott-AB @Thomas-debug-creator :

I'm currently in the process of updating our `openmct-hello` repository-- will update here once it's been completed. We will have some good example code for defining an openmct plugin within the next week or so. Thanks for your patience!

Cheers,
Jesse

**Comment by ozyx** [comment, marked as answer]
@AdamElliott-AB @Thomas-debug-creator

Good news, everyone!

I've published the first cut of the new and improved [`openmct-hello`](https://github.com/nasa/openmct-hello) repository (Open MCT Plugin Template). This provides a basic framework for developing and bundling an Open MCT plugin using webpack.

In it you will find:
- An example of a basic Vue 3 compatible View / ViewProvider for defining a stripped-down LAD Table component (Latest Available Data Table).
- A proper webpack configuration for development complete with source maps and HMR, using `webpack-dev-server`.
- A basic webpack configuration for production builds, which excludes `openmct` from the production bundle for distribution.

Please subscribe to the repo for updates, as we plan to add other examples which utilize more of the Open MCT API. We plan to add Views and ViewProviders for other frameworks as well, such as React. Please check it out and let us know if there are any questions or suggestions for improvement. 

Thanks again!
